### ISSUE TITLE
BSD-3-Clause isn't properly detected if the project name has spaces or slashes

### ISSUE THREAD

**Issue by webknjaz** [licensee/licensee#404]
### Describe the bug

If the third clause has a slash or spaces in the project name (like `Word1 Word2` or `some/thing`), `licensee` cannot match the license.

### Steps to reproduce the behavior

https://github.com/pypa/gh-action-pypi-publish/blame/57bdcc0/LICENSE.md#L17

```text
* Neither the name of pypa/gh-action-pypi-publish nor the names of its
  contributors may be used to endorse or promote products derived from
  this software without specific prior written permission.
```

For this chunk, `NOASSERTION` is returned but when I remove the slash, it's identified as BSD-3-Clause correctly

### Expected behavior

BSD-3-Clause should be returned

### Screenshots

N/A

### Additional context

N/A

**Comment by mlinksva** [comment]
Just FYI, substituting words in the body of a license is very prone to NOASSERTION.

There's no way spaces will be accommodated without a substantial change to how licensee works; a new matcher that understands license templates and what may be substituted for would be needed.

There's a possible easy fix for slashes though which I've opened #405 to potentially address.

**Comment by webknjaz** [comment]
Right. That's understandable.
